api/v1beta1/common_types.go

```diff
@@ -120,6 +120,10 @@ const (
 	// instead of being a source of truth for eventual consistency.
 	// This annotation can be used to inform MachinePool status during in-progress scaling scenarios.
 	ReplicasManagedByAnnotation = "cluster.x-k8s.io/replicas-managed-by"
+
+	// ChangeCauseAnnotation is the annotation set on MachineSets by users to identify the cause of revision changes.
+	// This annotation will be shown when users run `clusterctl alpha rollout history`.
+	ChangeCauseAnnotation = "cluster.x-k8s.io/change-cause"
```

Comment:
This annotation is never updated, right? So the cause is always "<none>". Is there a follow-up planned to update this annotation? 
If so, FYI, that work could potentially overlap with the in-place propagation work (https://github.com/kubernetes-sigs/cluster-api/issues/7731).

Reply:
I don't have the plan to update this specific annotation directly from the codes. I assume that this annotation is set by users, like [Kubernetes](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/).

Also, I'm okay with propagating this annotation set in a machinedeployment to machinsets within the scope of the label propagation as Kubernetes does a similar behavior (as described below), but I guess it shouldn't be specific for this annotation.

> CHANGE-CAUSE is copied from the Deployment annotation kubernetes.io/change-cause to its revisions upon creation. You can specify theCHANGE-CAUSE message by: